src/ArrayOfSets.hpp

```diff
@@ -133,38 +134,43 @@ class ArrayOfSets : protected ArrayOfSetsView< T, INDEX_TYPE, BUFFER_TYPE >
     ParentClass::assimilate( reinterpret_cast< ParentClass && >( src ) );
 
     INDEX_TYPE const numSets = size();
+    ArrayOfArraysView< T, INDEX_TYPE const, true, BUFFER_TYPE > const view =
+      ArrayOfArraysView< T, INDEX_TYPE const, true, BUFFER_TYPE >( numSets,
+                                                                   this->m_offsets,
+                                                                   this->m_sizes,
+                                                                   this->m_values );
+    BUFFER_TYPE< INDEX_TYPE > const sizes = this->m_sizes;
```

Comment:
There seems to be no way to manually "resize" a single sub-array through `view` after value deduplication. So I just capture `sizes` separately, seems to work well.

Reply:
This looks good to me.